src/mesh/parallel.jl

```diff
@@ -1,5 +1,8 @@
-# Partition mesh using a static domain decomposition algorithm based on leaf cell count alone
-function partition!(mesh::ParallelTreeMesh)
+# Partition mesh using a static domain decomposition algorithm
+# based on leaf cell count and tree structure.
+# If all children of a cell are leaves, the algorithm will keep them together on one rank
+# to allow for coarsening.
+function partition!(mesh)
```

Comment:
Also, I think we should find a way to allow both the old and the new way of partitioning. Thus please restore the old partition function such that we can use it if needed. Maybe in the future we can give a partitioning function as an argument to the mesh, such that users can also easily implement their own partitioning approach.

Reply:
I added a keyword argument `allow_coarsening=true` to switch back to the old version. With this set to `false`, the algorithm is identical to the old one.